### PR TITLE
HHH-10101 Implement nonstrict-read-write mode in Infinispan 2LC

### PR DESCRIPTION
**Pull request by rvansa**
JIRA: https://hibernate.atlassian.net/browse/HHH-10101
Depends on: https://github.com/hibernate/hibernate-orm/pull/1074

### CODE REVIEW

File: hibernate-infinispan/src/test/java/org/hibernate/test/cache/infinispan/functional/TombstoneTest.java [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+
+import static org.junit.Assert.*;
+
+
+/**
+ * Tests specific to tombstone-based caches
+ *
+ * @author Radim Vansa &lt;rvansa@redhat.com&gt;
+ */
+public class TombstoneTest extends AbstractNonInvalidationTest {
```
Comment (galderz) [inline hibernate-infinispan/src/test/java/org/hibernate/test/cache/infinispan/functional/TombstoneTest.java]: Very good addition, thanks for adding these tests.

**Comment by galderz** [comment]
@rvansa Looks good, one question though: did you add any functional tests showing how stale data is returned with non-strict? I've only noticed non-functional tests that verify that. I might have missed it :)

**Comment by rvansa** [comment]
@galderz Nope, it did not occur to me that we should test such not very desired behaviour. But I can add that VersionedTest yet.

**Comment by rvansa** [comment]
@galderz I have added tests that check that after DB commit but before tx commit, there is a stale value in the cache.
I have also changed the remove behaviour: the invalidation happens only after successful DB commit (initially I have invalidated it immediatelly, but if the transaction was rolled back, the entry could not be cached again) - so that's closer to what inserts and updates do.

**Comment by galderz** [comment]
Ok, makes sense. The tests look good too. Integrating....

**Comment by galderz** [comment]
@rvansa I'm testing this on master. It doesn't apply cleanly on 5.0.x.

**Comment by galderz** [comment]
Integrated in master.

**Comment by rvansa** [comment]
@galderz Okay, I'll check 5.0.x and create another PR for that.

**Comment by sebersole** [comment]
If something applies cleanly on master but not on 5.0 at this point I would
expect a problem with something not applied in both places.  There is still
very little variance between the 2 branches at this point.

**Comment by rvansa** [comment]
@galderz What was the problem when applying on 5.0? I had no issues, neither code conflict nor any failed tests. Here's the branch: https://github.com/rvansa/hibernate-orm/tree/HHH-10101_5.0

**Comment by sebersole** [comment]
Tomorrow is the 5.0.2 release.  Either this work should be resolved against 5.0 branch soon or the Jira should be updated to adjust the fixed version to 5.0.3

**Comment by rvansa** [comment]
Ack, thanks for the info. I'll poke Galder.

**Comment by galderz** [comment]
@rvansa Applied HHH-10101_5.0 cleanly and pushed. This should be ready now.

**Comment by dreab8** [comment]
should not be applied also to master?

**Comment by rvansa** [comment]
@dreab8 It was applied few weeks ago to master.

**Comment by dreab8** [comment]
@rvansa thanks I haven't noticed :-)

**Comment by sebersole** [comment]
So this was applied to 5.0 in time for 5.0.3 iiuc but the Jira was never resolved.  Meaning that we resolved 5.0.3 with this fix in place, but nothing about it in the release notes.

Guys, if you apply this PRs, can you please resolve the corresponding Jira?  Thanks

**Pull request closed without merging**